### ISSUE TITLE
Sublayers filtered by zoomlevel only

### ISSUE THREAD

**Issue by nebulon42** [systemed/glug#5]
According to the documentation something like this should be possible (see `on()`):
````
layer(:building, :zoom=>14..20, :source=>:osm_data) {
    fill_color _building_color

    on(16..20) {
        fill_outline_color _building_outline_color
    }
}
````

But it seems that Glug expects an additional filter condition here:
````
../glug/bin/glug:5:in `instance_eval': can't dup FalseClass (TypeError)
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:244:in `_set_filter'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:200:in `on'
	from (eval):34:in `block (2 levels) in <main>'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:89:in `instance_eval'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:89:in `layer'
	from (eval):31:in `block in <main>'
	from ../glug/bin/glug:5:in `instance_eval'
	from ../glug/bin/glug:5:in `block in <main>'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:71:in `instance_eval'
	from /var/lib/gems/1.9.1/gems/glug-0.0.2/lib/glug.rb:71:in `initialize'
	from ../glug/bin/glug:5:in `new'
	from ../glug/bin/glug:5:in `<main>'
````

If I specify e.g. `on(16..20, building=='yes')` the stylesheet compiles fine.

@systemed

**Comment by systemed** [comment]
Thanks for spotting! I've just pushed a fix for this, though haven't reissued the gem yet - do you want to have a try from source? You can do that with something like:

````ruby
require '../glug/lib/glug.rb'

puts Glug::Stylesheet.new {
	version 8
	name "my lovely stylesheet"
	# etc etc.
}

**Comment by nebulon42** [comment]
I can confirm that it works. Thanks!
Fixed in 0a20718963ae74420588d539a789d9541bfa2bc3.
